### ISSUE TITLE
dnf: Install packages together instead of one at a time

### ISSUE THREAD

**Issue by bowlofeggs** [ansible/ansible-modules-extras#642]
Hello!

I have a suggestion for a small improvement to the dnf module. When invoked with the `with_items` option and handed a list of packages, the dnf module will create one dnf process for each package, rather than giving one dnf package a list of packages to install. This takes longer, but it also can increase the chance of failing due to getting a broken Fedora mirror since it will communicate with more mirrors.

Here's some example YAML of what I mean:

```
- dnf: name={{ item }} state=present
  with_items:
      - bash-completion
      - vim-enhanced
```

I may be able to find some time to make a pull request for this soon, but I wanted to file an issue to document the request. Thanks!

**Comment by gregdek** [comment]
Thanks @rbarlow. Of course a patch would be welcome. :)

**Comment by abadger** [comment]
Also, the dnf module is in need of a rewrite.  If you'd like to contribute
to that, the work in progress PR is here: here:
https://github.com/ansible/ansible-modules-extras/pull/527

**Comment by abadger** [comment]
We've just merged a rewrite of the dnf module to the devel branch that may implement this https://raw.githubusercontent.com/ansible/ansible-modules-extras/devel/packaging/os/dnf.py .  Could you give the new code a try?  Either by checking out the devel branch from the git repo and giving it a try or copying just the rewritten module to your machine to override the one shipped with 1.9.x?

**Comment by abadger** [comment]
AFAICT from looking at the code this is now taken care of as all of the packages are first marked for install in one step and then the transaction is run as a second step.

Feel free to open a new issue if you find that the rewritten module is not doing this correctly.

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
